projects/core/components/dialog/test/dialog.component.spec.ts

```diff
@@ -0,0 +1,58 @@
+import {Component, DebugElement} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {TuiIdService} from '@taiga-ui/cdk';
+import {configureTestSuite, TuiPageObject} from '@taiga-ui/testing';
+
+import {TuiRootModule} from '../../root/root.module';
+import {TuiDialogModule} from '../dialog.module';
+import {TuiDialogService} from '../dialog.service';
+import {tuiDialogOptionsProvider} from '../dialog.tokens';
+
+describe(`Dialog with TUI_DIALOG_OPTIONS`, () => {
+    @Component({
+        template: `
+            <tui-root></tui-root>
+        `,
+    })
+    class TestComponent {}
+
+    const closeable = false;
+
+    let fixture: ComponentFixture<TestComponent>;
+    let tuiDialogService: TuiDialogService;
+    let pageObject: TuiPageObject<TestComponent>;
+
+    function getCloseButton(): DebugElement {
+        return pageObject.getByAutomationId(`tui-dialog__close`)!;
+    }
+
+    configureTestSuite(() => {
+        TestBed.configureTestingModule({
+            imports: [NoopAnimationsModule, TuiRootModule, TuiDialogModule],
+            declarations: [TestComponent],
+            providers: [
+                tuiDialogOptionsProvider({closeable}),
+                {
+                    provide: TuiDialogService,
+                    useFactory: () => new TuiDialogService(TestBed.inject(TuiIdService)),
+                },
```

Comment:
Two things that need your attention:
1. Since the abstract dialog service's constructor is not protected anymore, I created the dialog instances manually on the tests, so then I removed the constructor from `dialog.service.ts`. What do you think? This way, or just add constructors to each dialog implementation?
2. `mobile-dialog.service.ts` did not need this work around. I investigated a bit, and saw that the issue I mentioned before happens in testing because of code coverage functionality. Now this does not happen for Mobile Dialog, because `addon-mobile` package could be misconfigured. In `project.json` if you remove the `/src` from `sourceRoot`, the test for Mobile Dialog fails too. So at the moment there is no code coverage report for this package, I don't know if that is intentional or not. Please tell me if I need to modify anything